server/src/main/java/com/octopus/teamcity/opentelemetry/server/OTELHelper.java

```diff
@@ -0,0 +1,101 @@
+package com.octopus.teamcity.opentelemetry.server;
+
+import com.octopus.teamcity.opentelemetry.common.PluginConstants;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import jetbrains.buildServer.log.Loggers;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class OTELHelper {
```

Comment:
"Helper" classes always make me a little nervous. Often ends up being a bit of a grab bag of functions rather than a carefully structured object model.
Not sure I have a specific recommendation there - more something to ponder.

Reply:
I see this "Helper" being refactored in the future as the plugin gets larger and more modified. The idea was, all the opentelemetry related code go into the helper and the build code stay in the BuildListener.